linkml/generators/shaclgen.py

```diff
@@ -109,10 +109,11 @@ def prop_pv_literal(p, v):
                         prop_pv(SH.nodeKind, SH.IRI)
                     else:
                         prop_pv(SH.nodeKind, SH.BlankNode)
-                elif r in sv.all_types().values():
+                elif r in sv.all_types():
+                    prop_pv(SH.nodeKind, SH.Literal)
                     rt = sv.get_type(r)
                     if rt.uri:
-                        prop_pv(SH.datatype, rt.uri)
+                        prop_pv(SH.datatype, Literal(rt.uri))
```

Comment:
I don't think this is right - even though the value of sh:datatype is used for validating literals, it is not itself a literal

See:

https://www.w3.org/TR/shacl/#DatatypeConstraintComponent

example:

```turtle
sh:property [
		sh:path ex:age ;
		sh:datatype xsd:integer ;
	] .
```